wildfirestorage-spring/app/src/components/sharelinks/shareLinks.jsx

```diff
@@ -0,0 +1,187 @@
+import React, { useState, useEffect } from 'react';
+import { GoHistory, GoTrash } from 'react-icons/go';
+import SimpleModal from '../simpleModal/simpleModal';
+
+const ShareLinks = () => {
+    const [shareLinks, setShareLinks] = useState([{}]);
+    const [showModal, setShowModal] = useState(false)
+    const [content, setContent] = useState(null)
+    const [count, setCount] = useState(0)
+    const [currentPage, setCurrentPage] = useState(1)
+    const [pageButtons, setPageButtons] = useState([])
+
+    const getLinkData = async (limit, offset) => {
+        let url = "/api/share-link/?" + "limit=" + limit + "&offset=" + offset
+        const response = await fetch(url, {
+            method: "GET",
+            headers: {
+                "Content-Type": "application/json",
+                "Accept": "application/json",
+            },
+            credentials: "include",
+            redirect: "follow",
+        });
+        if (response.redirected) {
+            document.location = response.url;
+        }
+        let d = await response.json();
+        setShareLinks(d);
+    }
+
+    const getLinkCount = async () => {
+
+        const response = await fetch("/api/share-link/count", {
+            method: "GET",
+            headers: {
+                "Content-Type": "application/json",
+                "Accept": "application/json",
+            },
+            credentials: "include",
+            redirect: "follow",
+        });
+        if (response.redirected) {
+            document.location = response.url;
+        }
+        let d = await response.json();
+        setCount(d);
+        const pageSize = 100;
```

Comment:
If there is a properties file, pageSize can be moved there for easy configuration